openfisca_uk/variables/demographic/person.py

```diff
@@ -68,13 +69,53 @@ def formula(person, period, parameters):
 class is_child(Variable):
     value_type = bool
     entity = Person
-    label = u"Whether this person is a child"
+    label = u"Is a child"
     definition_period = YEAR
 
     def formula(person, period, parameters):
         return person("age", period) < 18
 
 
+class child_index(Variable):
+    value_type = int
+    entity = Person
+    label = u"Child reference number"
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        age = person("age", period)
+        child = person("is_child", period)
+        child_benunit = where(child, person.benunit.members_entity_id, -1)
+        child_ranking = (
+            pd.Series(age)
+            .groupby(child_benunit)
+            .rank(ascending=False, method="dense")
+            .astype(int)
+        )
+        adjusted_for_adults = where(
+            not_(person("is_child", period)), 100, child_ranking
+        )
+        return adjusted_for_adults
+
+
+class is_benunit_eldest_child(Variable):
+    value_type = bool
+    entity = Person
+    label = u"Eldest child in the benefit unit"
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
```

Comment:
If I'm reading this right, the goal of this and the prior function is to select the child with the highest age within each benefit unit, using the id to break ties, and deprioritizing people with `~is_child`. If so, I think this logic could be simplified to something like:
```
eldest_child_id = (
    df.sort_values(
        ["child", "age", "id"], 
        ascending=[False, False, False]
    ).groupby("bu")[["bu", "id"]]
    .head(1)
    .rename(columns=dict(id="eldest_child_id"))
)
```
Then either merging back to a DataFrame or using `transform` (haven't figured that one out) to get the desired bool vector. I gave this a shot in [this notebook](https://colab.research.google.com/drive/12SqmvdD60ZTiRo7vTsQvjQ0wVGd5fB00#scrollTo=ou482MJHRnrY).

Optional for now, could make it an issue for later, or ignore if I'm missing something.